js/interactive-guides/utils.js

```diff
@@ -82,27 +80,23 @@ var utils = (function() {
         //console.log("buttonName ", buttonName);
         return buttonName;
     };
-    
-    var __parseActionTag = function(strDesc) {
-        var resultStr;
-        if (strDesc.indexOf("<action") !== -1) {
-            var firstIndex = strDesc.indexOf("<action");
-            //console.log("1st index of <action> ", firstIndex);
-            var lastIndex = strDesc.lastIndexOf("</action>") + 9;
-            //console.log("last index of </action> ", lastIndex);
-            var origActionStr = strDesc.slice(firstIndex, lastIndex);
-            //console.log("original action ", origActionStr);
-            var name =  __getNameAction(origActionStr); 
+
+    var __parseActionTag = function(str) {
+        var resultStr = str;                    
+        var actionArray = str.match(/<action\b[^>]*>((\s|\S)*?)<\/action>/gm);
+        //console.log("parseStr: ", actionArray);
+        for (var a in actionArray) {
+            var origActionStr = actionArray[a];
+            //console.log("action[" + a + "]", origActionStr);
+            var name =  __getTitleAction(origActionStr); 
             if (name) {          
                 var callback = __getCallbackAction(origActionStr);
                 var buttonName = __getButtonName(origActionStr);
```

Comment:
In the existing getButtonName, you're looking for <b> and </b> which may not be there. Any call to indexOf has to check whether the index returning back is > -1.